source/grid/manifold_lib.cc

```diff
@@ -1149,6 +1149,145 @@ CylindricalManifold<dim, spacedim>::push_forward_gradient(
 
 
 
+// ============================================================
+// EllipticalManifold
+// ============================================================
+template <int dim, int spacedim>
+EllipticalManifold<dim, spacedim>::EllipticalManifold(
+  const Point<spacedim> &    center,
+  const Tensor<1, spacedim> &major_axis_direction,
+  const double               eccentricity)
+  : ChartManifold<dim, spacedim, spacedim>(
+      EllipticalManifold<dim, spacedim>::get_periodicity())
+  , direction(major_axis_direction)
+  , center(center)
+  , cosh_u(1.0 / eccentricity)
+  , sinh_u(std::sqrt(cosh_u * cosh_u - 1.0))
+{
+  // Throws an exception if dim!=2 || spacedim!=2.
+  Assert(dim == 2 && spacedim == 2, ExcNotImplemented());
+  // Throws a exception if eccentricity is not in range.
+  Assert(std::signbit(cosh_u * cosh_u - 1.0) == false,
+         ExcMessage(
+           "Invalid eccentricity: It must satisfy 0 < eccentricity < 1."));
+  const double direction_norm = direction.norm();
+  Assert(direction_norm != 0.0,
+         ExcMessage(
+           "Invalid major axis direction vector: Null vector not allowed."));
+  direction /= direction_norm;
+}
+
+
+
+template <int dim, int spacedim>
+std::unique_ptr<Manifold<dim, spacedim>>
+EllipticalManifold<dim, spacedim>::clone() const
+{
+  const double eccentricity(1.0 / cosh_u);
+  return std::make_unique<EllipticalManifold<dim, spacedim>>(center,
```

Comment:
The builds are currently failing because of missing C++14 support. Just write this as
```
std_cxx14::make_unique(...)
```
and make sure to include the header `deal.II/base/std_cxx14/memory.h` as well.